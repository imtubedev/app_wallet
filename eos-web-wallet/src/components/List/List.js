import * as React from "react";

type PropTypes = {
  /** list data */
  data: Array<any>,
  /** list item */
  renderItem: () => React.Node
};

// TODO abstract upper level to avoid unused dispatch
// TODO reject recusive lists
const List = ({ data, renderItem, ...props }: PropTypes) => (
  <ul {...props}>
    {data &&
      data.map(
        (item, key) =>
          item && Array.isArray(item)
            ? renderItem(item, key)
            : renderItem({ key, ...item })
      )}
  </ul>
);

export default List;
