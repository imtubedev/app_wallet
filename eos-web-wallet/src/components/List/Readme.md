A simple List example:

```js
let data = [
  { text: 'one' },
  { text: 'two' },
  { text: 'three' }
];

// The key is required on your renderItem
const renderItem = ({ text }, key) => (
  <li key={key}><div>{text}</div></li>
);

<div className="list-example">
  <List data={data} renderItem={renderItem} />
</div>
```
