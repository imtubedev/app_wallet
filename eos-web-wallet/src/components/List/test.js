import * as React from "react";
import renderer from 'react-test-renderer';
import List from './';

describe('<List />', () => {
  it('renders without crashing', () => {
    const data = [
      { text: 'One' },
      { text: 'Two' },
      { text: 'Three' },
    ];
    const renderItem = ({ key, text }) => (
      <p key={key}>{text}</p>
    );

    const tree = renderer.create(
      <List
        data={data}
        renderItem={renderItem}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})


