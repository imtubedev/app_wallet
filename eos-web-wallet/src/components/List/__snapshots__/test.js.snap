// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders without crashing 1`] = `
<ul>
  <p>
    One
  </p>
  <p>
    Two
  </p>
  <p>
    Three
  </p>
</ul>
`;
